cpp/src/arrow/util/hashing.h

```diff
@@ -233,20 +234,21 @@ class HashTable {
     assert(!*entry);
     entry->h = FixHash(h);
     entry->payload = payload;
-    ++n_filled_;
+    ++size_;
+
     if (NeedUpsizing()) {
-      // Resizing is expensive, avoid doing it too often
-      Upsize(size_ * 4);
+      DCHECK_OK(Upsize(capacity_ * kLoadFactor));
```

Comment:
Hmm, you removed the comment which explained the `* 4`. Since resizing is expensive we avoid doing it too often and therefore choose an upsize factor larger than the load factor.